src/pages/posts.js

```diff
@@ -1,22 +1,24 @@
 import { useEffect } from 'react';
 
-import { getAllPosts } from 'lib/posts';
+import { getPaginatedPosts } from 'lib/posts';
 
 import TemplateArchive from 'templates/archive';
 
-export default function Posts({ posts }) {
+export default function Posts({ posts, pagination }) {
   const title = 'All Posts';
   const slug = 'posts';
 
-  return <TemplateArchive title={title} posts={posts} slug={slug} />;
+  pagination.basePath = '/posts';
```

Comment:
we could probably put this in `getStaticProps` too right? 🤔  probably doesn't matte too much but just thinking out loud

Reply:
yeah, I thought the same. I can change it to be inside `getStaticProps`, something like this? (just to leave the function `getPaginatedPosts` without that responsability):

```
    props: {
      posts,
      pagination: {
        ...pagination,
        basePath = '/posts'
      },
    },
```

Reply:
yeahhh just thinking since most of the code is there makes sense to keep it together with it 🤷  but nbd either way